src/cpp/src/vlm_pipeline.cpp

```diff
@@ -341,7 +282,137 @@ ov::Tensor merge_text_and_image_embeddings_llava(
 
     return merged_embeds;
 }
+
+EncodedGenerationResult get_lm_encoded_results(
+    ov::InferRequest& language,
+    ov::InferRequest& embedding,
+    const ov::Tensor& inputs_embeds,
+    const VLMConfig& m_vlm_config,
+    const std::shared_ptr<StreamerBase>& streamer_ptr,
+    Sampler& sampler,
+    std::vector<SequenceGroup::Ptr> requests
+) {
+    SequenceGroup::Ptr request = requests.back();
+    GenerationHandle generation = std::make_shared<GenerationHandleImpl>(request->get_generation_stream(), request->get_sampling_parameters());
+
+    language.set_tensor("inputs_embeds", inputs_embeds);
+
+    size_t history_len = language.get_tensor("attention_mask").get_shape().at(1);
+    language.get_tensor("attention_mask").set_shape({1, history_len + inputs_embeds.get_shape()[1]});
+    std::fill_n(language.get_tensor("attention_mask").data<int64_t>(), language.get_tensor("attention_mask").get_size(), 1);
+
+    language.get_tensor("position_ids").set_shape({1, inputs_embeds.get_shape().at(1)});
+    std::iota(language.get_tensor("position_ids").data<int64_t>(), language.get_tensor("position_ids").data<int64_t>() + language.get_tensor("position_ids").get_size(), history_len);
+
+    language.get_tensor("beam_idx").set_shape({ BATCH_SIZE });
+    language.get_tensor("beam_idx").data<int32_t>()[0] = 0;
+
+    language.infer();
+
+    int64_t sequence_len = language.get_tensor("logits").get_shape().at(1);
+    request->schedule_tokens(sequence_len);
+
+    SamplerOutput sampler_output = sampler.sample(requests, language.get_tensor("logits"));
+
+    // logits include image and prompt embedings 
+    if (m_vlm_config.model_type == VLMModelType::LLAVA) {
+        request->update_processed_tokens_num(request->get_prompt_len());
```

Comment:
why is it llava specific?
typically, it's supposed to call finish_iteration() after each inference

Reply:
updated